### ISSUE TITLE
`failed to allocate memory` when continuously initializing `keras.applications` models in a loop.

### ISSUE THREAD

**Issue by Zhaopudark** [keras-team/keras#19471]
Hello. Thank you for your contribution and maintenance of Keras.

I'm using `pytest`  and parametrize-testing a customized loss function that uses a model from `keras.applications`.  When the combination of parameters became too many, I encountered an OOM problem of `failed to allocate memory`.  I did a little preliminary debugging and analyzing to locate the problem. I found it caused by continuously initializing `keras.applications` models in a loop. I have tried `keras.backend.clear_session()`, but it did not work.

A piece of code that reproduces the problem is as follows：
```python
import os
os.environ["KERAS_BACKEND"] = "tensorflow"
import keras
for i in range(200):
    keras.backend.clear_session()
    app = keras.applications.vgg16.VGG16()
    length = len(app.layers)
    del app
    print(length)
```

Additional information:
- Platform: WSL2
- Python: 3.12.2
- Keras: 3.1.1
- Backend: Tensorflow(GPU)2.16.1
- cuda：12.4
- cudnn：8.9.7
- GPU: NVIDIA GeForce RTX 4090

So, I have the following 3 points of confusion:
- Is it a bug or a feature?
- What core reasons cause the problem? Why is the memory consumed uncontrollably when continuously initializing `keras.applications` models in a loop? Why `keras.backend.clear_session()` did not work?
- How can I control the memory consumption and make it successfully to continuously initialize keras.applications models in a loop without the `failed to allocate memory` problem?

Thanks in advance.

**Comment by Zhaopudark** [comment]
@fchollet Thanks for your advice. `gc.collect()` can easily solve my problem as the following:
```python
import os
import gc
os.environ["KERAS_BACKEND"] = "tensorflow"
import keras
for i in range(200):
    gc.collect()
    app = keras.applications.vgg16.VGG16()
    length = len(app.layers)
    print(length)
```

**Comment by fchollet** [comment]
An update: we're adding `gc.collect()` to `clear_session()` , since wanting to reset memory usage is a common use case for it.
